### PR TITLE
OP-484 movement and lot fix

### PR DESCRIPTION
**Pull request by mwithi**
See OP-484.

### CODE REVIEW

File: src/main/java/org/isf/medicalstock/model/Lot.java [hunk 1]
```diff
@@ -34,13 +32,15 @@
 import javax.persistence.Id;
 import javax.persistence.JoinColumn;
 import javax.persistence.ManyToOne;
-import javax.persistence.OneToMany;
 import javax.persistence.Table;
 import javax.persistence.Transient;
 import javax.validation.constraints.NotNull;
 
+import org.hibernate.annotations.Formula;
```
Comment (dbmalkovsky) [inline src/main/java/org/isf/medicalstock/model/Lot.java:39]: This import is not used; please remove.

File: src/test/java/org/isf/medicalstockward/test/Tests.java [hunk 2]
```diff
@@ -36,6 +36,7 @@
 import org.isf.medicalstock.model.Lot;
 import org.isf.medicalstock.model.Movement;
 import org.isf.medicalstock.service.LotIoOperationRepository;
+import org.isf.medicalstock.service.MedicalStockIoOperations;
```
Comment (dbmalkovsky) [inline src/test/java/org/isf/medicalstockward/test/Tests.java:39]: This import is unused; please remove.

**Comment by dbmalkovsky** [comment]
In `org.isf.medicalstock.test.Tests` there are some remnants of `jpa` that can be removed (see attached patch file).   Once removed, `mvn clean test` runs fine:
````
Results :

Tests run: 1958, Failures: 0, Errors: 0, Skipped: 5
````
[RemoveJPA.patch.txt](https://github.com/informatici/openhospital-core/files/6428435/RemoveJPA.patch.txt)

After applying the patch and removing the unused imports it LGTM.

File: src/main/java/org/isf/medicalstock/model/Lot.java (hunk 1 above)
Comment (mwithi) [reply src/main/java/org/isf/medicalstock/model/Lot.java]: Done

File: src/test/java/org/isf/medicalstockward/test/Tests.java (hunk 2 above)
Comment (mwithi) [reply src/test/java/org/isf/medicalstockward/test/Tests.java]: Done

File: src/main/java/org/isf/medicalstock/model/Lot.java [hunk 3]
```diff
@@ -84,15 +84,50 @@
 	@Column(name="LT_COST")
 	private BigDecimal cost;
 
+	/**
+	 * Automatic calculated field for a lot's quantity stocked in the main store, 
+	 * taking in account only the main MedicalStock movements (charges and discharges).<br/>
```
Comment (dbmalkovsky) [inline src/main/java/org/isf/medicalstock/model/Lot.java:89]: DocLint reports that self-closing element is not allowed.   I think just replace it with `<br>`

File: src/main/java/org/isf/medicalstock/model/Lot.java [hunk 4]
```diff
@@ -84,15 +84,50 @@
...
+	 * @see <a href="https://github.com/informatici/openhospital-doc/blob/develop/doc_admin/AdminManual.adoc#5-1-19-internalpharmacies">Admin Manual</a>
+	 * @see <a href="https://github.com/informatici/openhospital-doc/blob/develop/doc_user/UserManual.adoc#63-pharmaceuticals-stock-ward-pharmaceuticals-stock-ward">User Manual</a>
+	 * </i>
+	 */
+	@Transient
+	private int mainStoreQuantity;
+	
+	/**
+	 * Automatic calculated field for a lot's quantity stocked in all wards, 
+	 * taking in account only the wards movements (inventories, discharges and transfers).<br/>
```
Comment (dbmalkovsky) [inline src/main/java/org/isf/medicalstock/model/Lot.java:107]: DocLint reports that self-closing element is not allowed.

File: src/main/java/org/isf/medicalstock/model/Lot.java [hunk 5]
```diff
@@ -84,15 +84,50 @@
...
+	 * <i>
+	 * @see <a href="https://github.com/informatici/openhospital-doc/blob/develop/doc_admin/AdminManual.adoc#5-1-19-internalpharmacies">Admin Manual</a>
+	 * @see <a href="https://github.com/informatici/openhospital-doc/blob/develop/doc_user/UserManual.adoc#63-pharmaceuticals-stock-ward-pharmaceuticals-stock-ward">User Manual</a>
+	 * </i>
+	 */
+	@Transient
+	private double wardsTotalQuantity;
+	
+	/**
+	 * Automatic calculated field for a overall lot's quantity (MedicalStock + MedicalStockWards).<br/>
```
Comment (dbmalkovsky) [inline src/main/java/org/isf/medicalstock/model/Lot.java:118]: DocLint reports that self-closing element is not allowed.

File: src/main/java/org/isf/medicalstockward/service/MedicalStockWardIoOperations.java [hunk 6]
```diff
@@ -120,6 +123,29 @@ public int getCurrentQuantityInWard(
 
 		return (int) (mainQuantity != null ? mainQuantity : 0.0);	
 	}
+	
+	/**
+	 * Gets the current quantity for the specified {@link Ward} and {@link Lot}.
+	 * @param ward - if {@code null} the quantity is counted for the whole hospital
+	 * @param lot - the {@link code}
```
Comment (dbmalkovsky) [inline src/main/java/org/isf/medicalstockward/service/MedicalStockWardIoOperations.java:130]: is `code` suppose to be `lot`?;   the symbol `code` cannot be resolved in this context.

File: src/test/java/org/isf/medstockmovtype/test/Tests.java [hunk 7]
```diff
@@ -73,26 +73,26 @@ public void testMovementTypeSets() throws Exception {
 	}
 
 	@Test
-	public void testIoGetMovementType() throws Exception {
+	public void testIofindOne() throws Exception {
```
Comment (dbmalkovsky) [inline src/test/java/org/isf/medstockmovtype/test/Tests.java:76]: should the method name be 'testIoFindOne()`?  Capital `F`.

File: src/test/java/org/isf/medstockmovtype/test/Tests.java [hunk 8]
```diff
@@ -73,26 +73,26 @@ public void testMovementTypeSets() throws Exception {
...
+		MovementType foundMovementType = medicalStockMovementTypeIoOperation.findOneByCode(code);
 		ArrayList<MovementType> movementTypes = medicalStockMovementTypeIoOperation.getMedicaldsrstockmovType();
 		assertThat(movementTypes.get(movementTypes.size() - 1).getDescription()).isEqualTo(foundMovementType.getDescription());
 	}
 
 	@Test
-	public void testIoGetMovementTypeNull() throws Exception {
-		assertThat(medicalStockMovementTypeIoOperation.getMovementType("notThere")).isNull();
+	public void testIofindOneNull() throws Exception {
+		assertThat(medicalStockMovementTypeIoOperation.findOneByCode("notThere")).isNull();
```
Comment (dbmalkovsky) [inline src/test/java/org/isf/medstockmovtype/test/Tests.java:85]: `testIoFindOneNull()` ?

File: src/test/java/org/isf/medstockmovtype/test/Tests.java [hunk 9]
```diff
@@ -114,23 +114,23 @@ public void testIoIsCodePresent() throws Exception {
...
+		MovementType foundMovementType = medicalStockMovementTypeIoOperation.findOneByCode(code);
 		boolean result = medicalStockMovementTypeIoOperation.deleteMedicaldsrstockmovType(foundMovementType);
 		assertThat(result).isTrue();
 		result = medicalStockMovementTypeIoOperation.isCodePresent(code);
 		assertThat(result).isFalse();
 	}
 
 	@Test
-	public void testMgrGetMovementType() throws Exception {
+	public void testMgrfindOne() throws Exception {
```
Comment (dbmalkovsky) [inline src/test/java/org/isf/medstockmovtype/test/Tests.java:125]: here to (last comment but there are others): `testMgrFindOne()`

File: src/main/java/org/isf/medicalstockward/service/MedicalStockWardIoOperations.java [hunk 10]
```diff
@@ -120,6 +123,29 @@ public int getCurrentQuantityInWard(
...
+	 * @return the total quantity.
+	 * @throws OHServiceException if an error occurs retrieving the quantity.
+	 */
+	public int getCurrentQuantityInWard(
+			Ward ward, 
+			Lot lot) throws OHServiceException 
+	{
+		Double quantity;
+		if (ward!=null)
```
Comment (dbmalkovsky) [inline src/main/java/org/isf/medicalstockward/service/MedicalStockWardIoOperations.java:139]: Formatting doesn't match standards; braces are not on their own line.

**Comment by dbmalkovsky** [comment]
Did not find anything significant; mostly cosmetic issues. All tests continue to pass. LGTM

File: src/main/java/org/isf/medicalstock/model/Lot.java (hunk 3 above)
Comment (mwithi) [reply src/main/java/org/isf/medicalstock/model/Lot.java]: Done

File: src/main/java/org/isf/medicalstockward/service/MedicalStockWardIoOperations.java (hunk 6 above)
Comment (mwithi) [reply src/main/java/org/isf/medicalstockward/service/MedicalStockWardIoOperations.java]: Yep! Fixed

File: src/test/java/org/isf/medstockmovtype/test/Tests.java (hunk 7 above)
Comment (mwithi) [reply src/test/java/org/isf/medstockmovtype/test/Tests.java]: Yep! Thanks!

File: src/main/java/org/isf/medicalstockward/service/MedicalStockWardIoOperations.java (hunk 10 above)
Comment (mwithi) [reply src/main/java/org/isf/medicalstockward/service/MedicalStockWardIoOperations.java]: Fixed

**Comment by mwithi** [comment]
> Did not find anything significant; mostly cosmetic issues. All tests continue to pass. LGTM

Thanks a lot!

**Pull request merged**